### PR TITLE
More structs and a generic client

### PR DESCRIPTION
**Pull request by radu-matei**
This PR adds events, jobs, and workers structs, and implements a generic client to be used by specific clients.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        &self,
+        opts: Option<ListOptions>,
+    ) -> Result<List<T>, Error> {
+        let url = format!("{}/v2/{}", self.rest.address, self.url_path);
+        let res = self.rest.req(Method::GET, &url, opts).send().await?;
+        let list: List<T> = serde_json::from_str(&res.text().await?)?;
+        Ok(list)
+    }
+
+    // Specific client might have selectors for listing clients.
```
Comment (radu-matei) [inline src/client.rs]: _specific clients_

File: src/client.rs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        url_path: String,
+        cfg: ClientConfig,
+        token: Option<String>,
+    ) -> Result<Self, Error> {
+        let rest = rest::Client::new(address, cfg, token)?;
+        Ok(Self { rest, url_path })
+    }
+}
+
+// #[async_trait::async_trait]
```
Comment (radu-matei) [inline src/client.rs]: Remove this, since we are no longer using an async trait.

File: src/client.rs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    meta::{List, ListOptions},
+    rest::{self, ClientConfig},
+};
+use anyhow::{Error, Result};
+use reqwest::{Method, RequestBuilder};
+use serde::{de::DeserializeOwned, Serialize};
+use serde_with::*;
+
+pub struct Client {
+    pub rest: rest::Client,
```
Comment (radu-matei) [inline src/client.rs]: Naming two separate structs `Client`, even if both are internal, is clunky.
I there is no need for both of them.

File: src/client.rs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        let rest = rest::Client::new(address, cfg, token)?;
+        Ok(Self { rest, url_path })
+    }
+}
+
+// #[async_trait::async_trait]
+impl Client {
+    pub async fn get<
+        T: Serialize + DeserializeOwned + Sized,
+        U: Serialize + DeserializeOwned + Sized,
```
Comment (radu-matei) [inline src/client.rs]: There is no need for two type parameters for a `get` operation.

**Comment by krancour** [comment]
@radu-matei I love watching you review your own PR. 😂 

Sorry I can't be more help with reviewing Rust code.

**Pull request merged**